### PR TITLE
[Improvement][Common] Optimize HashMap Initial Capacity

### PR DESCRIPTION
**Pull request by zhuangchong**
## What is the purpose of the pull request

Optimize HashMap Initial Capacity

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
      * @return check result code
      */
     public Map<String, Object> checkPageParams(int pageNo, int pageSize) {
-        Map<String, Object> result = new HashMap<>(2);
+        Map<String, Object> result = new HashMap<>(4);
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java:51]: In my opinion, this doesn't actually make much sense. No matter if the initial capacity is 2 or 4

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java [hunk 2]
```diff
@@ -103,7 +109,8 @@ public static boolean isEmpty(Collection coll) {
             return emptyMap;
         }
         String[] strings = str.split(separator);
-        Map<String, String> map = new HashMap<>(strings.length);
+        int initialCapacity = (int)(strings.length / DEFAULT_LOAD_FACTOR) + 1;
+        Map<String, String> map = new HashMap<>(initialCapacity);
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java:113]: keep old-style maybe better

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java (hunk 1 above)
Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java:51]: There is a difference. When the storage is 2 and the capacity is initialized to 2, automatic scaling of the HashMap is triggered

The HashMap load factor is 0.75 by default, the initialization capacity is 2, and the HashMap is automatically expanded when put the second element

I feel that can use the default capacity, or can specify the initial capacity, but can't specify an initial value that will trigger HashMap expansion in scenarios where storage is small

---

是有区别的，当存储为2并初始化容量为2会触发hashmap的自动扩容

hashmap负载因子是默认0.75,初始化容量为2，当put第二个元素的时候会触发hashmap自动扩容

我感觉可以使用默认容量，也可以指定初始容量，但是不能指定一个会触发hashmap扩容并在存储很小的场景下的一个初始值

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java (hunk 2 above)
Comment (zhuangchong) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java:113]: This change is to reduce the automatic scaling of the HashMap to get the maximum initial capacity based on the specified storage capacity and load factor.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/BaseController.java:51]: but..hashmap‘s min size is 16.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java (hunk 2 above)
Comment (CalvinKirs) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java:113]: you are right~

Comment (CalvinKirs) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/CollectionUtils.java:113]: you are right~

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**